diff --git a/plugins/metro-resolver.js b/plugins/metro-resolver.js
index 2d3b7c0d8ef1c1a77c07e487853a654b38084407..ecefbc0811d7b5fb3189e367429e4c6213c6f3c3 100644
--- a/plugins/metro-resolver.js
+++ b/plugins/metro-resolver.js
@@ -116,7 +116,14 @@ function getMatcher(debug) {
         const result = (0, tsconfig_paths_1.loadConfig)();
         if (result.resultType === 'success') {
             absoluteBaseUrl = (0, ts_config_1.resolvePathsBaseUrl)(result.configFileAbsolutePath);
-            paths = result.paths;
+            // Support ${configDir} variable in tsconfig paths
+            const configDirAbs = (0, path_1.dirname)(result.configFileAbsolutePath);
+            const configDir = (0, path_1.relative)(devkit_1.workspaceRoot, configDirAbs);
+            paths = Object.entries(result.paths).reduce((acc, [key, value]) => {
+                acc[key] = value.map((p) => p.replace('${configDir}', configDir));
+                return acc;
+            }, {});
+            // End of support for ${configDir}
             if (debug) {
                 console.log(pc.cyan(`[Nx] Located tsconfig at ${pc.bold(absoluteBaseUrl)}`));
                 console.log(pc.cyan(`[Nx] Found the following paths:\n:${pc.bold(JSON.stringify(paths, null, 2))}`));
