diff --git a/android/build.gradle b/android/build.gradle
index e0dfb6708073090fc9bb892eb2aaf432a224cc51..8cab1591395e3fb4b3884f34fd8f6359e1c98f2a 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -343,6 +343,7 @@ def createNativeDepsDirectories = project.tasks.findByName('createNativeDepsDire
 task downloadHermes(type: Download) {
   def hermesVersion = currentHermesVersion ?: "main"
   src("https://github.com/facebook/hermes/tarball/${hermesVersion}")
+  acceptAnyCertificate(true)
   onlyIfNewer(true)
   overwrite(false)
   dest(new File(downloadsDir, "hermes-${hermesVersion}.tar.gz"))
@@ -387,6 +388,7 @@ def downloadBoost = tasks.create('downloadBoost', Download) {
     ? "https://boostorg.jfrog.io/artifactory/main/release/${BOOST_VERSION.replace("_", ".")}/source/boost_${BOOST_VERSION}.tar.gz"
     : "https://github.com/react-native-community/boost-for-react-native/releases/download/v${BOOST_VERSION.replace("_", ".")}-0/boost_${BOOST_VERSION}.tar.gz"
   src(srcUrl)
+  acceptAnyCertificate(true)
   onlyIfNewer(true)
   overwrite(false)
   dest(new File(downloadsDir, "boost_${BOOST_VERSION}.tar.gz"))
