Index: libs/api/domains/syslumenn/src/lib/client/models/syslumennAuction.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libs/api/domains/syslumenn/src/lib/client/models/syslumennAuction.ts b/libs/api/domains/syslumenn/src/lib/client/models/syslumennAuction.ts
new file mode 100644
--- /dev/null	(revision 15164e7c901132c39c17c987513d9c40b001e515)
+++ b/libs/api/domains/syslumenn/src/lib/client/models/syslumennAuction.ts	(revision 15164e7c901132c39c17c987513d9c40b001e515)
@@ -0,0 +1,13 @@
+export interface ISyslumennAuction {
+  embaetti?: string
+  starfsstod?: string
+  tegund?: string
+  andlag?: string
+  dagsetning?: string
+  klukkan?: string
+  fastanumer?: string
+  andlagHeiti?: string
+  gerdarbeidendur?: string
+  gerdartholar?: string
+  lausafjarmunir?: string
+}
Index: libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts b/libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts
--- a/libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts	(revision 1ddc771d659bad35a65b7a7261d7838cd73abccd)
+++ b/libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts	(revision 624cc9cb412f6741f8632b5b3d864034b14f2f89)
@@ -1,5 +1,6 @@
 import { HttpService, Inject, Injectable } from '@nestjs/common'
 import { IHomestay } from './models/homestay'
+import { ISyslumennAuction } from './models/syslumennAuction'
 import { ILogin } from './models/login'
 import { Person, Attachment, DataUploadResponse } from '../models/dataUpload'
 import { constructUploadDataObject } from './models/dataUpload'
@@ -48,6 +49,23 @@
       .get(url)
       .toPromise()
 
+    return response.data
+  }
+
+  async getSyslumennAuctions(): Promise<ISyslumennAuction[] | null> {
+    await this.login()
+
+    const url = `${this.clientConfig.url}/api/Uppbod/${this.id}`
+
+    const headers = {
+      'Content-type': 'application/json',
+      Authorization: `Bearer ${this.accessToken}`,
+    }
+
+    const response: {
+      data: ISyslumennAuction[]
+    } = await this.httpService.get(url, { headers: headers }).toPromise()
+
     return response.data
   }
 
Index: libs/api/domains/syslumenn/src/lib/models/syslumennAuction.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libs/api/domains/syslumenn/src/lib/models/syslumennAuction.ts b/libs/api/domains/syslumenn/src/lib/models/syslumennAuction.ts
new file mode 100644
--- /dev/null	(revision c4dc9f7874799f7ce9840acc3788a6c7cb48aa53)
+++ b/libs/api/domains/syslumenn/src/lib/models/syslumennAuction.ts	(revision c4dc9f7874799f7ce9840acc3788a6c7cb48aa53)
@@ -0,0 +1,54 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+import { ISyslumennAuction } from '../client/models/syslumennAuction'
+
+@ObjectType()
+export class SyslumennAuction {
+  @Field({ nullable: true })
+  office?: string
+
+  @Field({ nullable: true })
+  location?: string
+
+  @Field({ nullable: true })
+  auctionType?: string
+
+  @Field({ nullable: true })
+  lotType?: string
+
+  @Field({ nullable: true })
+  lotName?: string
+
+  @Field({ nullable: true })
+  lotId?: string
+
+  @Field({ nullable: true })
+  lotItems?: string
+
+  @Field({ nullable: true })
+  auctionDate?: string
+
+  @Field({ nullable: true })
+  auctionTime?: string
+
+  @Field({ nullable: true })
+  petitioners?: string
+
+  @Field({ nullable: true })
+  respondent?: string
+}
+
+export const mapSyslumennAuction = (
+  auction: ISyslumennAuction,
+): SyslumennAuction => ({
+  office: auction.embaetti ?? '',
+  location: auction.starfsstod ?? '',
+  auctionType: auction.tegund ?? '',
+  lotType: auction.andlag ?? '',
+  lotName: auction.andlagHeiti ?? '',
+  lotId: auction.fastanumer ?? '',
+  lotItems: auction.lausafjarmunir ?? '',
+  auctionDate: auction.dagsetning ?? '',
+  auctionTime: auction.klukkan ?? '',
+  petitioners: auction.gerdarbeidendur ?? '',
+  respondent: auction.gerdartholar ?? '',
+})
Index: libs/api/domains/syslumenn/src/lib/syslumenn.resolver.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libs/api/domains/syslumenn/src/lib/syslumenn.resolver.ts b/libs/api/domains/syslumenn/src/lib/syslumenn.resolver.ts
--- a/libs/api/domains/syslumenn/src/lib/syslumenn.resolver.ts	(revision 1ddc771d659bad35a65b7a7261d7838cd73abccd)
+++ b/libs/api/domains/syslumenn/src/lib/syslumenn.resolver.ts	(revision c5c7570dadabc2127c1c6fe36d837f64168d840c)
@@ -1,6 +1,7 @@
 import { Args, Directive, Query, Resolver } from '@nestjs/graphql'
 import { GetHomestaysInput } from './dto/getHomestays.input'
 import { Homestay } from './models/homestay'
+import { SyslumennAuction } from './models/syslumennAuction'
 import { SyslumennService } from './syslumenn.service'
 
 const cacheTime = process.env.CACHE_TIME || 5
@@ -16,4 +17,10 @@
   getHomestays(@Args('input') input: GetHomestaysInput): Promise<Homestay[]> {
     return this.syslumennService.getHomestays(input.year)
   }
+
+  // Note: We don't cache the Auction data, as it's prone to changes only minutes before the auction takes place.
+  @Query(() => [SyslumennAuction])
+  getSyslumennAuctions(): Promise<SyslumennAuction[]> {
+    return this.syslumennService.getSyslumennAuctions()
+  }
 }
Index: libs/api/domains/syslumenn/src/lib/syslumenn.service.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/libs/api/domains/syslumenn/src/lib/syslumenn.service.ts b/libs/api/domains/syslumenn/src/lib/syslumenn.service.ts
--- a/libs/api/domains/syslumenn/src/lib/syslumenn.service.ts	(revision 1ddc771d659bad35a65b7a7261d7838cd73abccd)
+++ b/libs/api/domains/syslumenn/src/lib/syslumenn.service.ts	(revision 624cc9cb412f6741f8632b5b3d864034b14f2f89)
@@ -1,5 +1,9 @@
 import { SyslumennClient } from './client/syslumenn.client'
 import { Homestay, mapHomestay } from './models/homestay'
+import {
+  SyslumennAuction,
+  mapSyslumennAuction,
+} from './models/syslumennAuction'
 import { Injectable } from '@nestjs/common'
 import { Person, Attachment, DataUploadResponse } from './models/dataUpload'
 
@@ -13,6 +17,12 @@
     return (homestays ?? []).map(mapHomestay)
   }
 
+  async getSyslumennAuctions(): Promise<SyslumennAuction[]> {
+    const syslumennAuctions = await this.syslumennClient.getSyslumennAuctions()
+
+    return (syslumennAuctions ?? []).map(mapSyslumennAuction)
+  }
+
   async uploadData(
     persons: Person[],
     attachement: Attachment,
