diff --git a/node_modules/react-native-navigation/lib/ios/RNNButtonsPresenter.m b/node_modules/react-native-navigation/lib/ios/RNNButtonsPresenter.m
index c426fea..0b4a416 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNButtonsPresenter.m
+++ b/node_modules/react-native-navigation/lib/ios/RNNButtonsPresenter.m
@@ -82,14 +82,14 @@
     if ([side isEqualToString:@"left"]) {
         [self replaceCurrentButtons:self.viewController.navigationItem.leftBarButtonItems
                         withButtons:barButtonItems];
-        [self.viewController.navigationItem setLeftBarButtonItems:barButtonItems animated:animated];
+        [self.viewController.navigationItem setLeftBarButtonItems:barButtonItems animated:NO];
     }
 
     if ([side isEqualToString:@"right"]) {
         [self replaceCurrentButtons:self.viewController.navigationItem.rightBarButtonItems
                         withButtons:barButtonItems];
         [self.viewController.navigationItem setRightBarButtonItems:barButtonItems
-                                                          animated:animated];
+                                                          animated:NO];
     }
 }
 
