diff --git a/lib/cache.js b/lib/cache.js
index 9d636aa6aae253e3a3cc8d6473e01d660c6c1c3b..426c926362c59859116b51f3866ff30a18d0b287 100644
--- a/lib/cache.js
+++ b/lib/cache.js
@@ -134,6 +134,9 @@ function restoreCache(paths, primaryKey, restoreKeys, options, enableCrossOsArch
             if (typedError.name === ValidationError.name) {
                 throw error;
             }
+            else if (/Cache service responded with (?!404)\d{3}$/.test(error.message)) {
+                throw error;
+            }
             else {
                 // Supress all non-validation cache related errors because caching should be optional
                 core.warning(`Failed to restore: ${error.message}`);
